tests/lib/tools/filter-debug-output

```diff
@@ -0,0 +1,88 @@
+#!/usr/bin/env python3
+
+import argparse
```

Comment:
This is a great way to improve readability. Can you add a small unit test to show how it filters / not filters a short amount of example output.

I think that over time we will find weird edge cases and without tests it will be magically misbehaving.

One thing I worry about now is the internal buffering done by spread when a test is taking longer (but not failing). I don't quite know how that is handled in practice with the code below.

Reply:
Added a spread test. Based on the result logs in the ci I didn't see problems related to the internat buffering. I'll recheck the current results to make sure we are not loosing data.